android/ouinet/src/main/java/ie/equalit/ouinet/OuinetNotification.kt

```diff
@@ -0,0 +1,155 @@
+            return PendingIntent.getBroadcast(
+                context,
+                requestCode,
+                intent,
+                getFlags()
+            )
+        }
+    }
+
+    private fun getActivityPendingIntent(
+        activityName: String
+    ): PendingIntent {
+        Intent(context, Class.forName(activityName)).also { intent ->
+            intent.action = Intent.ACTION_MAIN
+            intent.addCategory(Intent.CATEGORY_LAUNCHER)
+            intent.addFlags(Intent.FLAG_ACTIVITY_NEW_TASK)
+            intent.addFlags(Intent.FLAG_ACTIVITY_REORDER_TO_FRONT)
+            intent.putExtra(FROM_NOTIFICATION_EXTRA, 1)
+            return PendingIntent.getActivity(
+                context,
+                HOME_CODE,
+                intent,
+                getFlags()
+            )
+        }
+    }
+
+    private val showConfirmCallback = Runnable {
+        try {
+            getServicePendingIntent(context, HIDE_CODE, config).send()
+        } catch (_: PendingIntent.CanceledException) {
+        }
+    }
+
+    fun create(showConfirmAction: Boolean, state: String): Notification {
+        var channelId = CHANNEL_ID
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
+            // Create a notification channel for Ouinet notifications. Recreating a notification
+            // that already exists has no effect.
+            val channel = NotificationChannel(
+                CHANNEL_ID,
+                config.channelName,
+                NotificationManager.IMPORTANCE_LOW)
+            channelId = channel.id
+            val notificationManager = context.getSystemService(NotificationManager::class.java)
+            notificationManager.createNotificationChannel(channel)
+        }
+        var contentTitle = "${config.title}"
+        if (!config.disableStatus) {
+            contentTitle += ": $state"
+        }
+        val stopPIntent = getBroadcastPendingIntent(STOP_CODE)
+        val homePIntent = getActivityPendingIntent(config.activityName!!)
+        val showConfirmPIntent = getServicePendingIntent(context, SHOW_CODE, config)
+        val confirmPIntent = getBroadcastPendingIntent(CONFIRM_CODE)
+        val notificationBuilder = NotificationCompat.Builder(context, channelId)
+            .setSmallIcon(config.statusIcon)
+            .setContentTitle(contentTitle)
+            .setContentText(config.description)
+            .setPriority(NotificationCompat.PRIORITY_DEFAULT)
+            .setContentIntent(stopPIntent)
+            .setAutoCancel(true) // Close on tap.
+            .addAction(config.homeIcon, config.homeText, homePIntent)
+            .addAction(config.clearIcon, config.clearText, showConfirmPIntent)
+        if (showConfirmAction) {
+            notificationBuilder
+                .addAction(config.clearIcon, config.confirmText, confirmPIntent)
+            handler.postDelayed(
+                showConfirmCallback,
+                3000 /* ms */
```

Comment:
Not sure if this is a good idea but wanted to share it with you anyway :smile_cat: 

What about replacing the comment `/* ms */` with a constant?
e.g. `3 * MILLISECONDS`

Reply:
I think it's a great idea. This `/* ms */` was a holder over from the original code from Ceno v1.6.x code, so I have no attachment to it, I'll replace with a constant.